src/ServerTrait.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Panther project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Panther;
+
+/**
+ * @author Dany Maillard <danymaillard93b@gmail.com>
+ *
+ * @internal
+ */
+trait ServerTrait
+{
+    private function keepServerOnTeardown(): void
+    {
+        PantherTestCase::$stopServerOnTeardown = false;
```

Comment:
I removed `startWebServer()` here because if there is in the user tests something like :
```php
public static function setUpBeforeClass()
{
    static::createPantherClient(['webServerDir' => '.');
}
```
The config `webServerDir` will not be used.

Reply:
Second point : calling the static property/method from the trait doesn't work, but from `PantherTestCase`, it works.

Reply:
Anyone using the listener/extension **must** remember that the server is **already running**, therefore stop it before starting a new one, so to me the server must be started anyway in the listener, it'll just be kept on teardown. But generally, changing server config between tests is not very common, the only case I might see is when one changes environment vars, but in such case, the listener would probably be a bad practice...

Reply:
My point was more about how to configure the web server : in your case, if we want to use the listener/extension, the only way to configure the web service would be only through the `PANTHER_WEB_SERVER_*` env vars, and I would like to allow also the configuration through `createPantherClient($option)`.

Reply:
Moreover, the listener/extension is working for all the user tests. So, if we want to run a `phpunit --filter UnitTest\\`, the web server will be launch whereas it's useless in this case.

Reply:
Using the listener/extension shouldn't really be fully compatible with manually configuring the server IMO :confused:

Reply:
Let's continue this topic after my last comment https://github.com/symfony/panther/pull/104#issuecomment-425613384 :)